include/envoy/ssl/handshaker.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include "envoy/api/api.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/network/transport_socket.h"
+#include "envoy/ssl/socket_state.h"
+
+#include "openssl/ssl.h"
+
+namespace Envoy {
+namespace Ssl {
+
+class HandshakerCallbacks {
+public:
+  virtual ~HandshakerCallbacks() = default;
+
+  /**
+   * Hands off the internally-held SSL object for external manipulation.
+   */
+  virtual bssl::UniquePtr<SSL> Handoff() PURE;
+  /**
+   * Accepts an SSL object for internal storage.
+   */
+  virtual void Handback(bssl::UniquePtr<SSL> ssl) PURE;
+  /**
+   * Called when a handshake is successfully performed.
+   */
+  virtual void LogHandshake(SSL* ssl) PURE;
+  /**
+   * Called when a handshake fails.
+   */
+  virtual void ErrorCb() PURE;
```

Comment:
Those 2 functions are mis-named after the existing implementations.

From the description, it sounds like they could be called `OnSuccessCb` and `OnFailureCb`?

Reply:
Done.